filesystem/src/main/java/org/frankframework/filesystem/IBasicFileSystem.java

```diff
@@ -105,7 +98,25 @@ public interface IBasicFileSystem<F> extends HasPhysicalDestination{
 	long getFileSize(F f) throws FileSystemException;
 	String getCanonicalName(F f) throws FileSystemException;
 	Date getModificationTime(F f) throws FileSystemException;
-
+	@Nullable
 	Map<String, Object> getAdditionalFileProperties(F f) throws FileSystemException;
 
+	/**
+	 * Safe method to get a string representing the canonical name of a file, or an error message
+	 * if the canonical name could not be established.
+	 *
+	 * Because this method is not guaranteed to return the actual canonical name, it should be used
+	 * only for error messages and logging.
+	 *
+	 * @param f File for which to try to get canonical name
+	 * @return Either the canonical name of the file, or an error.
+	 */
+	default String getCanonicalNameOrError(F f) {
```

Comment:
```suggestion
	default String getCanonicalNameOrErrorMessage(F f) {
```
Otherwise it might suggest that it gives a canonical name or throws an error.

Reply:
Ben ik met je eens, die twijfel had ik ook. Maar dit is wel heel lang. Ik dacht aan `getCanonicalNameForLog()` maar het is niet alleen voor logging, ook voor error messages, dus dat denkt het ook niet.
Eigenlijk vind ik alle namen voor deze method lelijk. Ik houd me open voor betere suggesties.

Reply:
`getCanonicalNameSafely` perhaps.